### PR TITLE
Task 7

### PR DESCRIPTION
**Pull request by duartenunes86**

### CODE REVIEW

File: __tests__/index.tests.js [hunk 1]
```diff
@@ -224,4 +225,55 @@ describe('error 404 GET /api/topics/', ()=>{
         .get('/api/articles/13/comments')
         .expect(200)
       })
-    })
\ No newline at end of file
+    })
+    describe("Post 201: inserts a new comment to some article", ()=>{ 
+      let dateCreated=""
```
Comment (EddTheDuck) [inline __tests__/index.tests.js]: What is this variable doing? Looks like it is not even being used?

File: __tests__/index.tests.js [hunk 2]
```diff
@@ -224,4 +225,55 @@ describe('error 404 GET /api/topics/', ()=>{
...
+        const newComment = {
+          username: 'butter_bridge',
+          body: 'I do not believe you'
+        };
+        return request(app)
+          .post('/api/articles/1/comments')
+          .send(newComment)
+          .expect(201)
+          .then((response) => {
+            console.log(response.body)
```
Comment (EddTheDuck) [inline __tests__/index.tests.js]: remove console.logs before submitting

File: __tests__/index.tests.js [hunk 3]
```diff
@@ -224,4 +225,55 @@ describe('error 404 GET /api/topics/', ()=>{
...
+            expect(response.status).toBe(201); // Check the HTTP status code
```
Comment (EddTheDuck) [inline __tests__/index.tests.js]: You do not need to assert this as this is what the .expect above is doing

File: __tests__/index.tests.js [hunk 4]
```diff
@@ -224,4 +225,55 @@ describe('error 404 GET /api/topics/', ()=>{
...
+    expect(response.body.comment.author).toEqual(newComment.username);
+    expect(response.body.comment.votes).toEqual(0)
+    expect(response.body.comment.article_id).toEqual(1)
+     
+     
+      })
+      
+        })
+        test('POST:404 on bad path', ()=>{
+
```
Comment (EddTheDuck) [inline __tests__/index.tests.js:272]: You do not need this test here and a bad path should already have been tested for in a previous ticket.

File: sion running out of my nose, pal! I'm the Sultan of Sentiment!                           [hunk 5]
```diff
@@ -0,0 +1,258 @@
+
```
Comment (EddTheDuck) [inline sion running out of my nose, pal! I'm the Sultan of Sentiment!                          ]: Don't know what this file is but it should defo be in your .gitignore so it's not uploaded to git!

File: __tests__/index.tests.js [hunk 6]
```diff
@@ -224,4 +225,55 @@ describe('error 404 GET /api/topics/', ()=>{
...
+          .expect(404)
+ 
+      })
+      test('POST:404 sends an appropriate and error message when given a valid but non-existent id', () => {
+        return request(app)
+          .post('/api/articles/999/comments')
+          .expect(404)
+          .then((response) => {
+            
+            expect(response.body.msg).toBe('item doesn\'t exist');
```
Comment (EddTheDuck) [inline __tests__/index.tests.js]: Is 'item doesn't exist' an appropriate error message in this case?

File: __tests__/index.tests.js [hunk 7]
```diff
@@ -224,4 +225,55 @@ describe('error 404 GET /api/topics/', ()=>{
...
+          .expect(201)
+          .then((response) => {
+            console.log(response.body)
+            expect(response.status).toBe(201); // Check the HTTP status code
+    expect(response.body.comment).toHaveProperty('created_at'); // Check if 'comment' property exists
+    expect(response.body.comment.body).toEqual(newComment.body);
+    expect(response.body.comment.author).toEqual(newComment.username);
+    expect(response.body.comment.votes).toEqual(0)
+    expect(response.body.comment.article_id).toEqual(1)
+
```
Comment (EddTheDuck) [inline __tests__/index.tests.js:266]: If you are having to have all these expect statments wouldn't it be worth using a different way of testing? e.g a more appropriate matcher.

Also you are using toEqual when checking for a number when should be toBe

File: __tests__/index.tests.js [hunk 8]
```diff
@@ -224,4 +225,55 @@ describe('error 404 GET /api/topics/', ()=>{
...
+      })
+      test('POST:400 sends an appropriate and error message when given an invalid id', () => {
+        return request(app)
+          .post('/api/articles/banana/comments')
+          .expect(400)
+          .then((response) => {
+            expect(response.body.msg).toBe('Invalid input');
+          });
+      }
+      )
```
Comment (EddTheDuck) [inline __tests__/index.tests.js:343]: Your tests aren 't complete. you are missing:
 Status 400, missing required field(s), e.g. no username or body properties
 Status 201, ignores unnecessary properties
Status 404, username does not exist

File: models/articles-models.js [hunk 9]
```diff
@@ -63,4 +63,21 @@ exports.selectArticles = () => {
...
       return result.rows;
-      })}})}
\ No newline at end of file
+      })}})}
+
+
+      exports.insertCommentToArticle = (username, body, id) => {
+        return checkExists('articles', 'article_id', id).then((article)=>{
+        if(article){
+          const date = new Date().toISOString()
```
Comment (EddTheDuck) [inline models/articles-models.js]: You do not need to add a date, this is done for you

File: models/articles-models.js [hunk 10]
```diff
@@ -63,4 +63,21 @@ exports.selectArticles = () => {
...
+      return db.query(`INSERT INTO comments(body,
+      votes,
+      author,
+      article_id,
+      created_at) VALUES($1,$2,$3,$4, $5) RETURNING *;`,[body, 0, username,id, date]).then((comment)=>{
```
Comment (EddTheDuck) [inline models/articles-models.js]: You also do not need to pass in number of votes

File: utils.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
+const db = require('./db/connection');
+const format = require('pg-format');
+
+
+
+exports.checkExists =  (table, column, value) => {
+  
+    const queryStr= format('SELECT * FROM %I WHERE %I = $1;', table, column)
+     return db.query(queryStr, [value]).then((dbOutput)=>{
+   console.log(db.Output)
```
Comment (EddTheDuck) [inline utils.js]: console.log

**Review by EddTheDuck — CHANGES_REQUESTED** [review]
A few changes needed here Duarte.

Also I noticed that a lot of the previous requested changes haven't been implemented. You need to spend some time going through all previous feedback and refactoring where necessary. 

Please do not submit another PR until all previous feedback has been implemented.

**Comment by duartenunes86** [comment]
Hi....I implemented previous feedback....I have been working hard to implement, it went back many times and they just approved it when it was implemented

**Pull request merged**